guake/utils.py

```diff
@@ -285,34 +283,29 @@ def set_final_window_rect(cls, settings, window):
 
     @classmethod
     def get_final_window_monitor(cls, settings, window):
-        """Gets the final screen number for the main window of guake."""
+        """Gets the final monitor for the main window of guake."""
 
-        screen = window.get_screen()
+        display = window.get_display()
 
         # fetch settings
         use_mouse = settings.general.get_boolean("mouse-display")
-        dest_screen = settings.general.get_int("display-n")
+        num_monitor = settings.general.get_int("display-n")
 
         if use_mouse:
+            pointer = display.get_default_seat().get_pointer()
+            if pointer is None:
+                monitor = display.get_primary_monitor()
+            else:
+                _, x, y = pointer.get_position()
+                monitor = display.get_monitor_at_point(x, y)
+        else:
+            monitor = display.get_monitor(num_monitor)
+            if monitor is None:
+                # monitor not found or num_monitor is wrong
+                # by default we use the primary monitor
+                monitor = display.get_primary_monitor()
 
-            # TODO PORT get_pointer is deprecated
-            # https://developer.gnome.org/gtk3/stable/GtkWidget.html#gtk-widget-get-pointer
-            win, x, y, _ = screen.get_root_window().get_pointer()
-            dest_screen = screen.get_monitor_at_point(x, y)
-
-        # If Guake is configured to use a screen that is not currently attached,
-        # default to 'primary display' option.
-        n_screens = screen.get_n_monitors()
-        if dest_screen > n_screens - 1:
-            settings.general.set_boolean("mouse-display", False)
-            settings.general.set_int("display-n", dest_screen)
-            dest_screen = screen.get_primary_monitor()
-
-        # Use primary display if configured
-        if dest_screen == ALWAYS_ON_PRIMARY:
```

Comment:
I believe the always on primary option in the "Appear on display" drop-down is for using the system configured primary display if chosen, while other options in the drop-down let the user override with a specific monitor. Looks like the explicit check is removed here in favor of a catch-all if the search for a specific monitor falls through. Not noticing adverse side effects from the removal of this, raising this in case any potential edge case comes to mind.

Reply:
Looked fine, just wanted to make sure we didn't accidentally kill any of the defined monitor options.